src/deadlock-detection/lockorder.cpp

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2019 Greg Griffith
+// Copyright (c) 2019 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "lockorder.h"
+#include "util.h"
+
+#ifdef DEBUG_LOCKORDER // this ifdef covers the rest of the file
+
+void CLockOrderTracker::potential_lock_order_issue_detected(const CLockLocation &thisLock,
+    const CLockLocation &otherLock,
+    const uint64_t &tid)
+{
+    LOGA("POTENTIAL LOCK ORDER ISSUE DETECTED\n");
```

Comment:
assertLockHeld(lot_mutex);

Reply:
we cannot use lock assertion macros within the deadlock detector code, it tends to break stuff or cause endless recursive looping. This is also the reason why this method is private so that we know if it is called we must already be holding the necessary lock.